crates/fuzzing/src/generators/api.rs

```diff
@@ -93,12 +90,15 @@ impl Arbitrary for ApiCalls {
         let mut scope = Scope::default();
         let max_rss = 1 << 30; // 1GB
 
-        // Calling an exported function of a `wasm-opt -ttf` module tends to
-        // take about 20ms. Limit their number to 100, or ~2s, so that we don't
-        // get too close to our 3s timeout.
-        let max_export_calls = 100;
+        // Total limit on number of API calls we'll generate. This exists to
+        // avoid libFuzzer timeouts.
+        let max_calls = 100;
 
         for _ in 0..input.arbitrary_len::<ApiCall>()? {
```

Comment:
I'm not too familiar with the `Arbitrary` trait, but is there perhaps a way that we can hook in here? Is there a way to indicate that we want at most a particular number of API calls?

Reply:
We have `int_in_range` but this draws from the front of the byte string, rather than the end, like lengths do. The mechanism that lengths use to get an integer in a range from the end of the byte string isn't a public API, unfortunately. (Reason lengths should be drawn from the end of the byte string: https://github.com/rust-fuzz/libfuzzer-sys/blob/0c450753/libfuzzer/utils/FuzzedDataProvider.h#L92-L97)

Overall, I don't think it really matters too much, though.

Reply:
Filed https://github.com/rust-fuzz/arbitrary/issues/36 for posterity